src/storage.ts

```diff
@@ -0,0 +1,77 @@
+import { DataStoreLevel, EventLogLevel, MessageStoreLevel, DwnConfig } from '@tbd54566975/dwn-sdk-js';
+import { SqliteDialect, MysqlDialect, PostgresDialect, Dialect, MessageStoreSql, DataStoreSql, EventLogSql } from '@tbd54566975/dwn-sql-store';
+
+import Database from 'better-sqlite3';
+import { createPool as MySQLCreatePool } from 'mysql2';
+import pg from 'pg';
+import Cursor from 'pg-cursor';
+
+const storageTypes = {
+  LEVEL    : 'level',
+  SQLITE   : 'sqlite',
+  MYSQL    : 'mysql',
+  POSTGRES : 'postgres'
+};
+
+function createDWNFromDB(db: Dialect): DwnConfig {
+  const messageStore = new MessageStoreSql(db);
+  const dataStore = new DataStoreSql(db);
+  const eventLog = new EventLogSql(db);
+  return { eventLog, dataStore, messageStore };
+}
+
+export function getDWNConfig(): DwnConfig {
+  // Initialize storage based on DWN_STORAGE environment variable
+  const storageURLstring = process.env.DWN_STORAGE || 'level://data';
+  const storageURL = new URL(storageURLstring);
+  const scheme = storageURL.protocol.slice(0, -1); // protocol always ends with ':'
+
+  switch(scheme) {
+  case storageTypes.LEVEL: {
+    const storageDir = storageURL.host + storageURL.pathname;
+    console.log('using leveldb storage in', storageDir);
+    const dataStore = new DataStoreLevel({ blockstoreLocation: storageDir + '/DATASTORE' });
+    const eventLog = new EventLogLevel({ location: storageDir + '/EVENTLOG' });
+    const messageStore = new MessageStoreLevel({
+      blockstoreLocation : storageDir + '/MESSAGESTORE',
+      indexLocation      : storageDir + '/INDEX'
+    });
+
+    return { eventLog, dataStore, messageStore };
+  }
+
+  case storageTypes.SQLITE: {
+    const db = new SqliteDialect({
+      database: async () => new Database(storageURL.host)
+    });
+
+    return createDWNFromDB(db);
+  }
+
+  case storageTypes.MYSQL: {
+    const db = new MysqlDialect({
+      pool: async () => MySQLCreatePool(storageURLstring),
+    });
+
+    return createDWNFromDB(db);
+  }
+
+  case storageTypes.POSTGRES: {
+    const db = new PostgresDialect({
+      pool   : async () => new pg.Pool({storageURL}),
+      cursor : Cursor,
+    });
+
+    return createDWNFromDB(db);
+  }
+
+  default: {
+    let schemes = [];
+    for (const [_, value] of Object.entries(storageTypes)) {
+      schemes.push(value);
+    }
+    throw 'Unknown storage protocol ' + scheme + '! Please use one of: ' + schemes.join(', ') + '. For details, see README';
+  }
+
+  }
+}
```

Comment:
I feel a little weird about repetition we're making with this calls. It's not critical at all, but how do you fell about something like this? Or somehow closer to this.

```suggestion
function createDWNFromDB({
  dataStoreConfig,
  eventLogConfig,
  messageStoreConfig,
}): DwnConfig {
  const messageStore = new MessageStoreSql(messageStoreConfig);
  const dataStore = new DataStoreSql(dataStoreConfig);
  const eventLog = new EventLogSql(eventLogConfig);

  return { eventLog, dataStore, messageStore };
}

export function getDWNConfig(): DwnConfig {
  // Initialize storage based on DWN_STORAGE environment variable
  const storageURLstring = process.env.DWN_STORAGE || 'level://data';
  const storageURL = new URL(storageURLstring);
  const scheme = storageURL.protocol.slice(0, -1); // protocol always ends with ':'
  let dataStoreConfig, eventLogConfig, messageStoreConfig;

  switch(scheme) {
  case storageTypes.LEVEL: {
    const storageDir = storageURL.host + storageURL.pathname;
    console.log('using leveldb storage in', storageDir);
    dataStoreConfig = { blockstoreLocation: storageDir + '/DATASTORE' };
    eventLogConfig= { location: storageDir + '/EVENTLOG' };
    messageStoreConfig = {
      blockstoreLocation : storageDir + '/MESSAGESTORE',
      indexLocation      : storageDir + '/INDEX'
    };

    break;
  }

  case storageTypes.SQLITE: {
    dataStoreConfig = eventLogConfig = messageStoreConfig = new SqliteDialect({
      database: async () => new Database(storageURL.host)
    });

    break;
  }

  case storageTypes.MYSQL: {
    dataStoreConfig = eventLogConfig = messageStoreConfig = new MysqlDialect({
      pool: async () => MySQLCreatePool(storageURLstring),
    });

    break;
  }

  case storageTypes.POSTGRES: {
    dataStoreConfig = eventLogConfig = messageStoreConfig = new PostgresDialect({
      pool   : async () => new pg.Pool({storageURL}),
      cursor : Cursor,
    });

    break;
  }

  default: {
    let schemes = [];
    for (const [_, value] of Object.entries(storageTypes)) {
      schemes.push(value);
    }
    throw 'Unknown storage protocol ' + scheme + '! Please use one of: ' + schemes.join(', ') + '. For details, see README';
  }

  }
  
  return createDWNFromDB({dataStoreConfig, eventLogConfig, messageStoreConfig});
}
```

Reply:
I'm not sure how this is better? it seems more complicated, but I don't really care one way or the other.

Reply:
that is fine, I just see a lot of repeated code and have an urge to change that 😉 

like in `case storageTypes.LEVEL:` there is basically the same code as in `createDWNFromDB` just with different arguments.

and each of the following cases makes a `return createDWNFromDB(db);` which should apply to basically every case and generally should be a return result of this function, hence I suggested moving it to the end and call once.